outputs.tf

```diff
@@ -25,12 +25,12 @@ output "runner_agent_role_name" {
 
 output "runner_role_arn" {
   description = "ARN of the role used for the docker machine runners."
-  value       = aws_iam_role.docker_machine.arn
+  value       = aws_iam_role.docker_machine[0].arn
```

Comment:
This will not work for the case you try to run the example for docker, e.g. a non docker-machine execturor. Did you check your change (actual deployment with a single instance)? The problem is that in that case the list is null, and you refer here in the output to the elment 0. 

Simple trick is to use a splat expression, add a default element (if null) and take the first element. 

An example looks like:

```hcl

output "runner_role_arn" {
  description = "ARN of the role used for the docker machine runners."
  value = element(concat(aws_iam_role.docker_machine.*.arn, [""]), 0)
}

```